docs/explanation/index.md

```diff
@@ -2,14 +2,10 @@
 
 These guides explain how Jubilant works and how it's developed.
 
-
-## Design goals
-
-When designing Jubilant, we had several important goals.
-
```

Comment:
I couldn't figure out a way to keep this wording but also have a document that wasn't "design goals" in the table of contents, and I don't think we want multiple ToCs on the page.

I'm happy to put this back if someone can suggest a way that it can work with other explanation docs being included as well.

Reply:
Thanks! How's this?